server/handler/pull_request_review.go

```diff
@@ -39,13 +40,64 @@ func (h *PullRequestReview) Handle(ctx context.Context, eventType, deliveryID st
 		return errors.Wrap(err, "failed to parse pull request review event payload")
 	}
 
+	pr := event.GetPullRequest()
+	repo := event.GetRepo()
+	owner := repo.GetOwner().GetLogin()
+	number := event.GetPullRequest().GetNumber()
 	installationID := githubapp.GetInstallationIDFromEvent(&event)
+
+	client, err := h.NewInstallationClient(installationID)
+	if err != nil {
+		return err
+	}
+
+	v4client, err := h.NewInstallationV4Client(installationID)
+	if err != nil {
+		return err
+	}
+
 	ctx, _ = h.PreparePRContext(ctx, installationID, event.GetPullRequest())
 
-	return h.Evaluate(ctx, installationID, common.TriggerReview, pull.Locator{
-		Owner:  event.GetRepo().GetOwner().GetLogin(),
-		Repo:   event.GetRepo().GetName(),
-		Number: event.GetPullRequest().GetNumber(),
-		Value:  event.GetPullRequest(),
+	mbrCtx := NewCrossOrgMembershipContext(ctx, client, owner, h.Installations, h.ClientCreator)
+	prctx, err := pull.NewGitHubContext(ctx, mbrCtx, client, v4client, pull.Locator{
+		Owner:  owner,
+		Repo:   repo.GetName(),
+		Number: number,
+		Value:  pr,
 	})
+	if err != nil {
+		return err
+	}
+
+	fc := h.ConfigFetcher.ConfigForPR(ctx, prctx, client)
+
+	evaluator, err := h.Base.ValidateFetchedConfig(ctx, prctx, client, fc, common.TriggerReview)
+	if err != nil {
+		return err
+	}
+	if evaluator == nil {
+		return nil
+	}
+
+	reviewState := pull.ReviewState(event.GetReview().GetState())
+	if !h.affectsApproval(reviewState, fc.Config.ApprovalRules) {
+		return nil
+	}
+
+	_, err = h.Base.EvaluateFetchedConfig(ctx, prctx, client, evaluator, fc)
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
#476 also adds a new function that we'll need to call here to make sure things like review requests and review dismissal still happen.